src/receiver-ui.py

```diff
@@ -0,0 +1,119 @@
+import ipaddress, subprocess
+from tkinter import *
+from tkinter import filedialog, messagebox
+from receiver import Receiver
+
+
+def register():
+    receiver.display_name = display_name_entry.get()
+    receiver.serial_number = serial_number_entry.get()
+    return_message = receiver.register()
+    if return_message == "Decoder already registered!":
+        messagebox.showerror("Error", return_message)
+    else:
+        messagebox.showinfo("Info", return_message)
+
+
+def receive():
+    ip = ip_entry.get()
+    port = port_entry.get()
+    if not is_valid_ip(ip):
+        messagebox.showerror("Error", "Invalid IP address.")
+        return
+    if not is_valid_port(port):
+        messagebox.showerror("Error", "Invalid port.")
+        return
+    subprocess.Popen(
+        [
+            "ffplay",
+            "-autoexit",
```

Comment:
It looks like the `-autoexit` flag is not working, and we need to manually close the window opened by FFMPEG. We should open a bug for that if we don't fix it before merging this PR.

Reply:
This has been an annoyance since the beginning unfortunately. I think this might be a bug on ffmpeg's side. I'll most likely open a bug like you suggested so we can find a solution